consensus/miner_signalled_consensus.md

```diff
@@ -0,0 +1,159 @@
+prevents diverging the block chain in case a large fraction of the miners have
+not upgraded their nodes to a version able to follow the new consensus protocol.
+
+## Specification
+
+### Overview
+
+In a chain fork, the new consensus protocol becomes active at a determined
+height if at least a determined majority of the key blocks in a determined
+"signalling" block interval in such chain fork contains a determined value in
+the header field `info`.
+
+The "signalling" block interval has a determined number of confirmations before
+the determined height at which the new consensus protocol may become
+active. This is in order to prevent the node from having to aggregate the
+results of the "signalling" block interval in a too short time interval.
+
+The value in the key block header field `info`:
+- Uses the whole field `info` i.e. is not a bitmask.
+- Is meant to be a pseudorandom number, so to limit the risk of the same value
+  being used by simultaneous competing consensus protocol proposals.
+- Is part of the validation rules of the new consensus protocol on the
+  "signalling" block interval.
+
+### Parameters
+
+In order for the miner-signalled consensus protocol upgrade to succeed, each
+node must have the same set of parameters.
+
+Assumptions:
+- Let us call C1 the current consensus protocol version, and H1 the height at
+  which it was first effective.
+
+#### New consensus protocol version
+
+The proposed new consensus protocol version.
+
+It must be strictly greater than C1 (strictly monotonically increasing consensus
+protocol version).
+
+#### Signal
+
+The "signal" value in the block header field `info`.
+
+It should be pseudorandom.
+
+#### Signalling block interval start
+
+The height at which the "signalling" block interval starts -
+included. Let us call it HS.
+
+HS should cater for the time required by most willing miners to adopt a version
+of the node able to signal and able to follow the new consensus protocol e.g. at
+least two weeks.
+
+HS must be strictly greater than H1.
+
+#### Signalling block interval end
+
+The height at which the "signalling" block interval ends -
+excluded. Let us call it HE.
+
+HE must be strictly greater than HS.
+
+The difference between HE and HS is recommended to be approximately a
+week i.e. 3360 (`= 7 * 24 * 20`).
+
+#### Proposed new consensus protocol height
+
+The height at which the new consensus protocol may be active. Let us call it H2.
+
+H2 should cater for the time required by the node for aggregating the results of
+the "signalling" block interval.
+
+H2 must be strictly greater than HE.
+
+The difference between H2 and HE is recommended to be approximately two
+hours i.e. 40 (`= 2 * 20`).
+
+#### Majority
+
+The number of key blocks in the "signalling" block interval that is meant to
+activate the new consensus protocol.
+
+It must be positive.
+
+It is recommended to be 90% of (HE - HS). So it is 3024 (that is - 90%
+of 3360) if (HE - HS) is 3360.
```

Comment:
Is 90% going to be hardcoded or it's going be a configuration option?

Reply:
Configuration option.